### PR TITLE
[PowerPC] Update data layout aligment of i128 to 16

### PR DESCRIPTION
**Pull request by lei137**
Fix 64-bit PowerPC part of https://github.com/llvm/llvm-project/issues/102783.

### CODE REVIEW

**Comment by brad0** [comment]
cc @nikic

**Review by amy-kwan — COMMENTED** [review]
For the most part, I think it looks good but I just have one comment about a change that (I think) doesn't need to happen.

File: clang/lib/Basic/Targets/OSTargets.h [hunk 1]
```diff
@@ -473,7 +473,7 @@ class LLVM_LIBRARY_VISIBILITY PS3PPUTargetInfo : public OSTargetInfo<Target> {
     this->IntMaxType = TargetInfo::SignedLongLong;
     this->Int64Type = TargetInfo::SignedLongLong;
     this->SizeType = TargetInfo::UnsignedInt;
-    this->resetDataLayout("E-m:e-p:32:32-Fi64-i64:64-n32:64");
+    this->resetDataLayout("E-m:e-p:32:32-Fi64-i64:64-i128:128-n32:64");
```
[the remark this answers is not in the document]
Comment (lei137) [reply clang/lib/Basic/Targets/OSTargets.h:476]: PS3's triple is `powerpc64-lv2`

[the remark this answers is not in the document]
Comment (RolandF77) [reply clang/lib/Basic/Targets/OSTargets.h:476]: Sony Cell had a PPU processor which was a PPC and some SPU processors which were not.

[the remark this answers is not in the document]
Comment (lei137) [reply clang/lib/Basic/Targets/OSTargets.h:476]: This class is specific for powerpc64 PPU, triple `powerpc64-lv2`, based on it's macro defines on line 461->465. Am I missing something or do we not want to update this specifically for PPC PS3 target for some reason?

In `llvm/lib/Target/PowerPC/PPCTargetMachine.cpp` I also see the comment:
```
The PS3 (OS Lv2) is a PPC64 machine with 32 bit
```

**Comment by lei137** [comment]
ping

File: clang/lib/Basic/Targets/OSTargets.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (RolandF77) [reply clang/lib/Basic/Targets/OSTargets.h:476]: Was just a clarification. It's the PPU so it's fine.

**Review by RolandF77 — APPROVED** [review]
LGTM

**Review by amy-kwan — APPROVED** [review]
Thanks for clarifying my earlier comment. LGTM.

**Pull request merged**